Project/rules.py

```diff
@@ -1,22 +1,72 @@
 import json
+import os
 
 class GameRules:
     """Loads game difficulty settings from JSON."""
-    def __init__(self, difficulty="Normal"):
+    def __init__(self, difficulty="Normal", player_role="Warrior"):
```

Comment:
I have some problem with this might be might pathing of my OS. 
But I would suggest allowing the user to pass a custom path for flexibility and testability

```suggestion
def __init__(self, difficulty="Normal", player_role="Warrior", filename="data/game_rules.json"):
    self.difficulty = difficulty
    self.player_role = player_role
    self.filename = filename
    self.load_rules()

```